core/chains/evm/types/models.go

```diff
@@ -168,11 +168,14 @@ func (h *Head) ChainHashes() []common.Hash {
 }
 
 func (h *Head) LatestFinalizedHead() commontypes.Head[common.Hash] {
-	for h != nil && !h.IsFinalized {
+	for h != nil {
+		if h.IsFinalized {
+			return h
+		}
 		h = h.Parent
```

Comment:
We made this change so that we can explicitly return nil in the case where there is no finalized head in the chain, see this discussion for more details: https://chainlink-core.slack.com/archives/C03ANJB9VB7/p1715163095224569?thread_ts=1714656712.938949&cid=C03ANJB9VB7

Reply:
We only replace the current head with its parent if it isn't finalized, and return the head if it is finalized. We are iterating through the chain to find a finalized head hence why we need to check the parent if the current head isn't finalized